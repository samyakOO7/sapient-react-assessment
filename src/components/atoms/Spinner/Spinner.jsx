import * as React from "react";
import CircularProgress from "@mui/material/CircularProgress";
import Box from "@mui/material/Box";
import { styled } from "@mui/system";

const Spinner = () => (
        <SpinnerContainer data-testid="spinner-box">
          <CircularProgress sx={{ color: "red" }} />
        </SpinnerContainer>
);

const SpinnerContainer = styled(Box)({
    display: "flex",
    justifyContent: "center",
    alignItems: "center"
});

export default Spinner;
