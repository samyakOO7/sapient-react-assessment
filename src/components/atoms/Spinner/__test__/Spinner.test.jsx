import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Spinner from "../Spinner";

describe("Spinner component", () => {
    it("renders without crashing", () => {
        const { getByTestId } = render(<Spinner />);
        const spinnerBox = getByTestId("spinner-box");
        expect(spinnerBox).toBeInTheDocument();
    });

    it("renders CircularProgress component", () => {
        const { getByTestId } = render(<Spinner />);
        const circularProgress = getByTestId("spinner-box");
        expect(circularProgress).toBeInTheDocument();
    });

    it("matches snapshot", () => {
        const { container } = render(<Spinner />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
