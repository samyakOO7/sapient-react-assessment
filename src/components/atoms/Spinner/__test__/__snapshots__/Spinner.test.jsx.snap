// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component matches snapshot 1`] = `
<div
  class="MuiBox-root css-1clwou3"
  data-testid="spinner-box"
>
  <span
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-wbsq5o-MuiCircularProgress-root"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-19t5dcl-MuiCircularProgress-circle"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </span>
</div>
`;
